### DISCUSSION TITLE
2.4.1

### DISCUSSION THREAD

**Discussion by blacs30** [blacs30/bitwarden-alfred-workflow#117] [Announcements]
## Changelog

* (#116) adds the option to use `apikey` login, this is needed in some cases where the `bw cli` requires otherwise a captcha to login (see Changes to CLI authentication? (Additional authentication required) bitwarden/cli#383)
* Make all default workflow commands start with `.bw` again
* disable auto sync, just add an item which can trigger the sync - this should remove some annoyances and allows to have all trigger keywords back under `.bw`
* unlock is now supporting complex passwords - Bitwarden cli 1.19 or newer is required

<hr /><em>This discussion was created from the release <a href='https://github.com/blacs30/bitwarden-alfred-workflow/releases/tag/2.4.1'>2.4.1</a>.</em>

**Comment by jinkiie22** [comment]
Thank you for your wonderful work and keeping it updated, it is much appreciated. 
I am getting the following error when trying to copy a password from the workflow: "runtime error: slice bounds out of range [:-1]" any idea what could be wrong? It copes URL and username just fine.

Running OSX Monterey; MBP m1 PRO.

**Reply by blacs30** [reply]
Thanks for the nice words and sorry to hear that copying passwords doesn't work.

Internally passwords are decrypted at runtime of the workflow whereas the username and url are cached from the Bitwarden CLI items output (and the cache is stored encrypted but differently and easier accessible for the workflow) 

Have you used the workflow successfully before, if yes which version(s)?
This runtime error would typically appear if the Bitwarden encryption key is a very old one. So you might be a long term Bitwarden user and therefore most likely have a "old" version of the encryption key at one point in time, in workflow version 2.2.0 the decryption flow was changed.

The solution to this scenario is to follow the [Bitwarden documentation for the rotation of the encryption key](https://bitwarden.com/help/article/account-encryption-key/). A logout of all devices happens after that.
